source/agora/node/Ledger.d

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+
+    Contains supporting code for tracking the current ledger.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.node.Ledger;
+
+import agora.common.Block;
+import agora.common.Data;
+import agora.common.Hash;
+import agora.common.Transaction;
+
+/// Ditto
+class Ledger
+{
+    /// data storage for all the blocks
+    private Block[Hash] ledger;
+
+    /// latest block height
+    private Block last_block;
+
+    /// Ctor
+    public this ()
+    {
+        auto block = getGenesisBlock();
+        this.addNewBlock(block);
+    }
+
+    /***************************************************************************
+
+        Called when a new transaction is received.
+
+        Creates a new block containing the transaction,
+        and adds the block to the ledger.
+
+        Params:
+            tx = the received transaction
+
+    ***************************************************************************/
+
+    public void receiveTransaction (Transaction tx) @trusted
```

Comment:
That's very network-oriented. What about loading from disk ? Will the name stick ?

Reply:
Hmm I mean I could name it `addTransaction`, but I was worried about this name because the transaction might be invalid, therefore it will **not** be added, despite the name.

So what do we call it?

Reply:
Damn hard questions. We can go with this now and change later.